### PR TITLE
Flexible additional metadata for MyDSpace items

### LINKED ISSUE #1591: Display additionnal metadata in DSpace 7 MyDSpace Workflow tasks list
By default, DSpace displays for each record in MyDSpace Workflow tasks list limited descriptive metadata: title, date, author and abstracts. 

The objective is to offer flexibility to repository managers to choose and display additionnal metadata in MyDSpace Workflow tasks dashboard being it another descriptive metadata or a costum administrative metadata (ex. notes to describe a problem with the document). 

Additionnal metadata feature would be activated has a specific configuration that can be turn off and on at will.  

The additionnal metadata can be displayed after descritive metadata block used by default by DSpace as illustrated below :  

additionnal-meta

The additionnal metadata will be displayed only to eperson that have worflow roles or are admin, but not to submitters.

### PR DESCRIPTION
**Pull request by nibou230**
## References
* Fixes #1591

## Description
Added a configuration section to insert one or multiple metadata at the end of an item description shown in the MyDSpace section.

## Instructions for Reviewers

List of changes in this PR:
* Added the `myDSpace` configuration section to list all additional metadata.
* Added a new section in both `item-detail-preview.component.spec.ts` and `item-list-preview.component.html` to display every metadata present in the configuration.
* Updated existing unit tests to ensure the new information is displayed properly.

**To display additional values, you can set the configuration to something like:**
```
myDSpace:
  additionalMetadataFields:
    - dc.identifier.uri
    - dc.description.provenance
```

It is only an example, you can set any value(s) you want at this point. By default, no additional metadata will be displayed.

_Note that private metadata (dc.description.provenance) will only be displayed for the administrators, as intended._

## Checklist
- [X] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [X] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [X] My PR doesn't introduce circular dependencies
- [X] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [X] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [X] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/config/my-dspace.interface.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Config } from './config.interface';
+
+/**
+ * Config that determines some aspects of the MyDSpace section
+ */
+export interface MyDSpaceConfig extends Config {
+  /**
+   * The list of additional metadatas to display at the end of the item component
+   */
+  additionalMetadatas: AdditionalMetaConfig[]
```
Comment (ybnd) [inline src/config/my-dspace.interface.ts:10]: "metadata" is already plural, `additionalMetadataFields` would be more clear

File: src/config/my-dspace.interface.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+/**
+ * Config that represents an additional metadata.
+ * Contains a value matching a metadata code.
+ */
+export interface AdditionalMetaConfig extends Config {
+  value: string;
```
Comment (ybnd) [inline src/config/my-dspace.interface.ts:18]: If you just make it a list of strings you can skip the `.map(m => m.value)` (would also make the config more readable)

**Comment by tdonohue** [comment]
@nibou230 : When you have a chance, could you rebase this on the latest `main` code?  It looks like this PR hasn't been updated in a long time, so it's trying to run old Node 12 test (which we no longer require on `main`) that result in the red x (❌ ) on this PR.

**Comment by nibou230** [comment]
Hi @tdonohue, do any weird behaviours are still happening for the cypress integration tests between 14.x and 16.x?

**Comment by tdonohue** [comment]
@nibou230 : Yes, unfortunately, we still sometimes get random failures with cypress integration tests. I've just restarted your Node 14 build in this PR as it looks like one of those random failures

**Comment by davide-negretti** [comment]
Hi @tdonohue @nibou230 
I've been working for DSpace CRIS 7 on an extension of this PR, which brings the same functionality on the search results. We could port that on DSpace 7 when this PR will have been merged.

**Comment by tdonohue** [comment]
@nibou230 : After merging #1791 (which is a larger, but important refactor), this now has some small merge conflicts.  I'd appreciate it if you could rebase this again when you get the chance.

**Comment by tdonohue** [comment]
Closing this PR as the developers have let me know this won't be possible to move forward.  The code remains here if anyone should wish to pick it back up again in the future.

**Pull request closed without merging**